### ISSUE TITLE
PA tower question

### ISSUE THREAD

**Issue by npoulos** [SoftFever/OrcaSlicer#433]
When you generate a PA tower from the calibration menu, the GCODE it generates uses the following to change the value.

M900 K0.054 L1000 M10

I'm just wondering what the M10 does? I've had a google and M10 seems to not be 3d printer specific.

I've been having a lot of problems trying to get reliable readings out of the tower so just wondering if maybe the M10 could be causing an issue?

Also wondering why L is set to 1000? Is that for a specific reason/what does it achieve?

Thanks.

**Comment by SoftFever** [comment]
@npoulos The `L` and `M` parameters are Bambu's only things.
I assume you selected the Bambu machine since you can see this.

Bambu's firmware is close-sourced, and no public documentation currently explains its parameters.

Nevertheless, this is what we know:
Unlike Marlin and Klipper, it uses a non-linearity equation/model to compensate for the extrusion. `L` stands for linearity. The model degenerates to a linear model when L is 1000.
M stands for volumetric speed. This parameter has no impact when the model is linear. For the non-linear model, it needs to test PA multiple times at different volumetric speeds. Imagine that it's a polynomial regression process.

In short, we set L1000 to ensure we want to `linear` advance. The M value is just an arbitral value since it's irrelevant in this case.

This is just a speculation based on the [comment](https://github.com/bambulab/BambuStudio/issues/348#issuecomment-1318384882) from their employees, it may not be 100% correct.
